### PR TITLE
✨ Update to cnspec v9

### PR DESCRIPTION
**Pull request by czunker**

### CODE REVIEW

**Comment by czunker** [comment]
I tried different setups.

Building a GCP VM image with upstream config from my home dir:
```
    mondoo-gcp-ubuntu-2004-secure-base.googlecompute.ubuntu2004: ✕ Fail:  C  50  Ensure events that modify user/group information are collected
    mondoo-gcp-ubuntu-2004-secure-base.googlecompute.ubuntu2004: 
    mondoo-gcp-ubuntu-2004-secure-base.googlecompute.ubuntu2004: 
    mondoo-gcp-ubuntu-2004-secure-base.googlecompute.ubuntu2004: Scanned 1 assets
    mondoo-gcp-ubuntu-2004-secure-base.googlecompute.ubuntu2004: 
    mondoo-gcp-ubuntu-2004-secure-base.googlecompute.ubuntu2004: Ubuntu 20.04.4 LTS
    mondoo-gcp-ubuntu-2004-secure-base.googlecompute.ubuntu2004:     C mondoo-gcp-ubuntu-2004-secure-base-20231010101357
    mondoo-gcp-ubuntu-2004-secure-base.googlecompute.ubuntu2004: 
    mondoo-gcp-ubuntu-2004-secure-base.googlecompute.ubuntu2004: See more scan results and asset relationships on the Mondoo Console: https://console.mondoo.com/space/fleet/2WZGv2eZzsPytYo2wlqHJFtzAK3?spaceId=charming-almeida-134558
    mondoo-gcp-ubuntu-2004-secure-base.googlecompute.ubuntu2004:
```

Building docker image with upstream config from my home dir:
```
    mondoo-docker-ubuntu-2204-secure-base.docker.ubuntu: ✓ Pass:  A 100  Ensure secure permissions on /etc/group are set
    mondoo-docker-ubuntu-2204-secure-base.docker.ubuntu: 
    mondoo-docker-ubuntu-2204-secure-base.docker.ubuntu: 
    mondoo-docker-ubuntu-2204-secure-base.docker.ubuntu: Scanned 1 assets
    mondoo-docker-ubuntu-2204-secure-base.docker.ubuntu: 
    mondoo-docker-ubuntu-2204-secure-base.docker.ubuntu: Ubuntu 22.04.3 LTS
    mondoo-docker-ubuntu-2204-secure-base.docker.ubuntu:     C determined_agnesi
    mondoo-docker-ubuntu-2204-secure-base.docker.ubuntu: 
    mondoo-docker-ubuntu-2204-secure-base.docker.ubuntu: See more scan results and asset relationships on the Mondoo Console: https://console.mondoo.com/space/fleet/2WZLaoRGueAGftdDIV9KZXZeJL5?spaceId=charming-almeida-134558
    mondoo-docker-ubuntu-2204-secure-base.docker.ubuntu:
```

Building a docker image with incognito mode:
```
==> mondoo-docker-ubuntu-2204-secure-base.docker.ubuntu: Running cnspec packer provisioner by Mondoo (Version: 9.0.2, Build: dev)
    mondoo-docker-ubuntu-2204-secure-base.docker.ubuntu: detected packer container image build
    mondoo-docker-ubuntu-2204-secure-base.docker.ubuntu: no configuration provided
    mondoo-docker-ubuntu-2204-secure-base.docker.ubuntu: scan packer build in incognito mode
    mondoo-docker-ubuntu-2204-secure-base.docker.ubuntu: scan completed successfully
    mondoo-docker-ubuntu-2204-secure-base.docker.ubuntu: Asset: keen_knuth
    mondoo-docker-ubuntu-2204-secure-base.docker.ubuntu: -----------------
```

**Comment by czunker** [comment]
@scottford-io / @chris-rock Could you please give this a try?

**Comment by czunker** [comment]
Fixes #129

File: .github/workflows/pr-test-lint.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ on:
       - 'go.sum'
 
 env:
-  GO_VERSION: '1.20'
+  GO_VERSION: '1.21'
```
Comment (chris-rock) [inline .github/workflows/pr-test-lint.yaml:12]: nit: i think we should make this consistent with cnquery [and use the ](https://github.com/mondoohq/cnquery/blob/main/.github/env)

File: examples/packer-docker/docker-ubuntu.pkr.hcl [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# Copyright (c) Mondoo, Inc.
```
Comment (chris-rock) [inline examples/packer-docker/docker-ubuntu.pkr.hcl:1]: nice addition

File: go.mod [hunk 3]
```diff
@@ -1,212 +1,107 @@
...
 replace github.com/slack-go/slack v0.12.1 => github.com/imilchev/slack v0.0.0-20230324120548-5380d7dd00a5
 
+require go.mondoo.com/cnquery/v9 v9.0.2
+
+require go.mondoo.com/cnspec/v9 v9.0.2
+
 require (
-	github.com/cockroachdb/errors v1.9.1
+	github.com/cockroachdb/errors v1.11.1
 	github.com/hashicorp/hcl/v2 v2.14.1
```
Comment (chris-rock) [inline go.mod:15]: nit: when we update those anyway we should use the opportunity to get to latest and greatest

File: provisioner/provisioner.go [hunk 4]
```diff
@@ -528,7 +524,7 @@ func (p *Provisioner) executeCnspec(ui packer.Ui, comm packer.Communicator) erro
 	var err error
 	if p.config.Incognito {
 		ui.Message("scan packer build in incognito mode")
-		scanService := scan.NewLocalScanner()
+		scanService := scan.NewLocalScanner(scan.WithRecording(providers.NullRecording{}))
```
Comment (chris-rock) [inline provisioner/provisioner.go:527]: Why do we need to set those? By default this should not record

File: go.mod [hunk 5]
```diff
@@ -1,212 +1,107 @@
 module go.mondoo.com/packer-plugin-cnspec
 
-go 1.19
+go 1.21
+
+toolchain go1.21.1
 
 replace github.com/slack-go/slack v0.12.1 => github.com/imilchev/slack v0.0.0-20230324120548-5380d7dd00a5
 
+require go.mondoo.com/cnquery/v9 v9.0.2
```
Comment (chris-rock) [inline go.mod:9]: nit: I recommend writing this as:

```
require (
	go.mondoo.com/cnquery/v9 v9.0.2
	go.mondoo.com/cnspec/v9 v9.0.2
)
```

**Review by chris-rock — APPROVED** [review]
This worked great. I tested it with GCP and Docker.

File: go.mod (hunk 5 above)
Comment (scottford-io) [reply go.mod]: @chris-rock I talked with @vjeffrey and asked whether we should bump the go.mod to v9.2.0. Do we not want the latest? 

Also, I updated the example templates and README to reference the latest tag of the plugin.

File: provisioner/provisioner.go (hunk 4 above)
Comment (czunker) [reply provisioner/provisioner.go:527]: Before this, I got a panic because of a nil reference.

File: go.mod (hunk 5 above)
Comment (czunker) [reply go.mod]: Sure, we can update this. v9.0.2 was the latest version as of the creation of this PR.

File: .github/workflows/pr-test-lint.yaml (hunk 1 above)
Comment (czunker) [reply .github/workflows/pr-test-lint.yaml]: I changed it

File: go.mod (hunk 3 above)
Comment (czunker) [reply go.mod]: Done

File: provisioner/provisioner.go (hunk 4 above)
Comment (czunker) [reply provisioner/provisioner.go:527]: @chris-rock The discussed follow-up https://github.com/mondoohq/packer-plugin-cnspec/pull/143

**Pull request merged**